src/ax_interface/socket_io.py

```diff
@@ -37,10 +59,52 @@ def __init__(self, mib_table, run_event, loop):
             try:
                 logger.info("Attempting AgentX socket bind...".format())
 
-                connection_routine = self.loop.create_unix_connection(
-                    protocol_factory=lambda: AgentX(self.mib_table, self.loop),
-                    path=constants.AGENTX_SOCKET_PATH,
-                    sock=self.ax_socket)
+                # Open the connection to the Agentx socket, we check the socket string to 
+                # either open a tcp socket or a Unix domain socket
+                if '/' in self.ax_socket_path:
```

Comment:
>if '/' in self.ax_socket_path: [](start = 16, length = 30)

As document ([http://www.net-snmp.org/docs/man/snmpd.html](http://www.net-snmp.org/docs/man/snmpd.html)) said, ipx may contains '/'. I think you will hit an exception.

Reply:
yes and there are a bunch of other ones that I saw, I'm going to do a specific function to deal with it